### PR TITLE
Features branch ready for deployment in development

### PR DESCRIPTION
**Pull request by tigamadou**
1. Linters installed
2. Webpack and loaders installed as development dependencies
3. HTML skeleton with content div element added
4. Created Home Menu and Contact Tabs using JavaScript only
5. Tabbed browsing installed
6. Each tab contents put in a separate module
7. Tab-switching logic done inside of index.js

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import './assets/sass/style.sass';
+import Home from './components/Home';
+import Menu from './components/Menu';
+import Contact from './components/Contact';
+
+class App {
```
Comment (MauricioRobayo) [inline src/index.js:6]: I have to say I am not in love with this class.

There is a lot of duplicate code, mostly when you create elements, give them a class and an innerHTML, and append some children.

If you had a helper that could handle all of that you could remove a lot of duplicate code here, something like this comes to my mind but probably you can do better:

```js
const createElement = ({
  type = 'div',
  textContent = '',
  classList = [],
  children = [],
}) => {
  const el = document.createElement(type);
  el.textContent = textContent;
  el.classList.add(...classList);
  el.append(...children);
  return el;
};
```

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    this.navElement1.className = 'nav-element';
+    this.navElement1.innerHTML = 'Home';
+
+    this.navElement2 = document.createElement('li');
+    this.navElement2.className = 'nav-element';
+    this.navElement2.innerHTML = 'Menu';
+
+    this.navElement3 = document.createElement('li');
+    this.navElement3.className = 'nav-element';
+    this.navElement3.innerHTML = 'Contact';
```
Comment (MauricioRobayo) [inline src/index.js:63]: You are basically doing the same three times 🤔 ... remember DRY?

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      this.renderContent();
+    });
+
+    this.navElement3.addEventListener('click', () => {
+      this.activeMenu = 'Contact';
+      this.navElement1.classList.remove('active');
+      this.navElement2.classList.remove('active');
+      this.navElement3.classList.add('active');
+      this.renderContent();
+    });
```
Comment (MauricioRobayo) [inline src/index.js:104]: Maybe if you use the `event` to get the `textContent` of the selected item  you could avoid repeating the same logic.

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    if (this.activeMenu === 'Home') {
+      this.content.innerHTML = '';
+      this.content.appendChild(Home());
+    } else if (this.activeMenu === 'Menu') {
+      this.content.innerHTML = '';
+      this.content.appendChild(Menu());
+    } else {
+      this.content.innerHTML = '';
+      this.content.appendChild(Contact());
+    }
```
Comment (MauricioRobayo) [inline src/index.js:117]: You could simplify this a bit if you take advantage of the fact that your functions are using the same names as the `activeMenu`:

```js
renderContent() {
  const content = { Home, Menu, Contact };
  this.content.innerHTML = '';
  this.content.append(content[this.activeMenu]());
}
```

File: src/components/Menu.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    const price = document.createElement('div');
+    price.className = 'price';
+    price.innerHTML = `$${Math.round(Math.random() * 100)}`;
+    const title = document.createElement('div');
+    title.className = 'title';
+    title.innerHTML = 'Lorem ipsum dolor sit amet consectetur adipisicing elit.';
+
+    menu.appendChild(image);
+    menu.appendChild(price);
+    menu.appendChild(title);
```
Comment (MauricioRobayo) [inline src/components/Menu.js:29]: There is a newer alternative to `appendChild` to simplify this.

<details>
 <summary>Hint</summary>
https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/append

```js
element.append(child1, child1, child3, ...)
```
</details>

File: src/components/Home.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  const description2 = document.createElement('p');
+  description2.innerHTML = 'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dignissimos ab sequi quo adipisci quos excepturi quidem praesentium, dicta fuga eveniet molestias, temporibus numquam voluptate necessitatibus totam aspernatur, aut ut labore!.';
+
+  const description3 = document.createElement('p');
+  description3.innerHTML = 'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dignissimos ab sequi quo adipisci quos excepturi quidem praesentium, dicta fuga eveniet molestias, temporibus numquam voluptate necessitatibus totam aspernatur, aut ut labore!.';
+
+  element.appendChild(title);
+  element.appendChild(description);
+  element.appendChild(description2);
+  element.appendChild(description3);
```
Comment (MauricioRobayo) [inline src/components/Home.js:18]: You could extract the duplicate code into a function and then use that function:

```js
function description() {
  const description = document.createElement('p');
  description.textContent = 'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dignissimos ab sequi quo adipisci quos excepturi quidem praesentium, dicta fuga eveniet molestias, temporibus numquam voluptate necessitatibus totam aspernatur, aut ut labore!.';
  return description;
}

export default function Home() {
  const element = document.createElement('div');
  const title = document.createElement('h2');
  title.textContent = 'Welcome';
  const paragraphsCount = 3;
  element.append(title, ...Array.from({ length: paragraphsCount }, description));

  return element;
}
```

**Review by MauricioRobayo — APPROVED** [review]
Great execution, the CSS transitions are splendid work. You nailed it, all requirements are met. Get out of here, move on to the next project!

Status: **Approved**.

All comments are *OPTIONAL*, they are just for your consideration.

**Comment by tigamadou** [comment]
@MauricioRobayo Thank you very much for the feedback. I'll work on them and refactor the code.

**Pull request merged**